### PR TITLE
Add version.yaml processing for Symfony/Yaml v4 support

### LINKED ISSUE #13: Support newer versions of symfony/yaml
We should add a preprocessor to YAML files to prevent the breaking changes in newer versions of symfony/yaml from affecting the ecosystem (i.e. unquoted string keys that could be interpreted as numbers no longer work in newer versions).

Refs: octobercms/october#5271

### PR DESCRIPTION
**Pull request by bennothommo**
Fixes https://github.com/wintercms/winter/issues/13
Follows on from https://github.com/wintercms/storm/pull/16

Moves the `version.yaml` processing into the System module and updates all version file parsing to use this new processor.

### CODE REVIEW

**Comment by LukeTowers** [comment]
As discussed on Discord we should probably actually make a single processor to deal with all of the incompatibilities, and apply it by default in the Storm library.

**Comment by bennothommo** [comment]
Follow up PR here: https://github.com/wintercms/storm/pull/18

File: modules/system/classes/VersionYamlProcessor.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    {
+        $lines = preg_split('/[\n\r]+/', $text, -1, PREG_SPLIT_NO_EMPTY);
+
+        foreach ($lines as $num => &$line) {
+            // Surround array keys with quotes if not already
+            $line = preg_replace_callback('/^\s*([\'"]{0}[^\'"\n\r:]+[\'"]{0})\s*:/m', function ($matches) {
+                return '"' . trim($matches[1]) . '":';
+            }, rtrim($line));
+
+            // Add quotes around any unquoted text proceeding an array key
```
Comment (mjauvin) [inline modules/system/classes/VersionYamlProcessor.php]:
```suggestion
            // Add quotes around any unquoted text preceeding an array key
```

**Comment by LukeTowers** [comment]
Is this PR still required with wintercms/storm#18 in play?

**Comment by mjauvin** [comment]
Pretty sure it is still required @LukeTowers .

It's doing extra stuff special to `version.yaml`

**Comment by LukeTowers** [comment]
What is the extra stuff exactly?

**Comment by mjauvin** [comment]
https://github.com/wintercms/winter/blob/b736df2b88d92f669c7c8b193b8a9d5781f803a7/modules/system/classes/VersionYamlProcessor.php#L29,L43

**Comment by bennothommo** [comment]
@LukeTowers as @mjauvin said, it is still required. Our usage of `!!!` to mark important updates also fails the YAML spec (in the way that we use it) . I didn't worry about it for the global parser because not only was it unlikely to be in use, if it was in use as a parsed value like the spec says, it would be intentional.

File: modules/system/classes/VersionYamlProcessor.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    {
+        $lines = preg_split('/[\n\r]+/', $text, -1, PREG_SPLIT_NO_EMPTY);
+
+        foreach ($lines as $num => &$line) {
+            // Surround array keys with quotes if not already
+            $line = preg_replace_callback('/^\s*([\'"]{0}[^\'"\n\r:]+[\'"]{0})\s*:/m', function ($matches) {
+                return '"' . trim($matches[1]) . '":';
+            }, rtrim($line));
+
+            // Add quotes around any unquoted text preceeding an array key
```
Comment (LukeTowers) [inline modules/system/classes/VersionYamlProcessor.php]:
```suggestion
            // Add quotes around any unquoted text following an array key
            // specifically to ensure usage of !!! in unquoted comments does not fail
```

**Pull request merged**